contracts/src/CrossChainApplications/GETTING_STARTED.md

```diff
@@ -0,0 +1,214 @@
+# Getting Started with an Example Teleporter Application
+
+This section walks through how to build an example cross-chain application on top of the Teleporter protocol, recreating the `ExampleCrossChainMessenger` contract that sends arbitrary string data from one chain to another. Note that this tutorial is meant for education purposes only. The resulting code is not intended for use in production environments.
+
+## Step 1: Create Initial Contract
+
+Create a new file called `MyExampleCrossChainMessenger.sol` in the directory that will hold the application.
+
+At the top of the file define the Solidity version to work with, and import the necessary types and interfaces.
+
+```solidity
+pragma solidity 0.8.18;
+
+import {ITeleporterMessenger, TeleporterMessageInput, TeleporterFeeInfo} from "../../Teleporter/ITeleporterMessenger.sol";
+import {ITeleporterReceiver} from "../../Teleporter/ITeleporterReceiver.sol";
+```
+
+Next, define the initial empty contract.
+
+```solidity
+contract MyExampleCrossChainMessenger {}
+```
+
+## Step 2: Integrating Teleporter Messenger
+
+Now that the initial empty `MyExampleCrossChainMessenger` is defined, it's time to integrate the `ITeleporterMessenger` that will provide the functionality to deliver cross chain messages.
+
+Create a state variable of `ITeleporterMessenger` type called `teleporterMessenger`. Then create a constructor for our contract that takes in an address where the Teleporter Messenger would be deployed on this chain, and set our state variable with it.
+
+```solidity
+contract ExampleCrossChainMessenger {
+    ITeleporterMessenger public immutable teleporterMessenger;
+
+    constructor(address teleporterMessengerAddress) {
+        teleporterMessenger = ITeleporterMessenger(teleporterMessengerAddress);
+    }
+}
+```
+
+## Step 3: Send and Receive
+
+Now that `MyExampleCrossChainMessenger` has an instantiation of `ITeleporterMessenger`, the next step is to add in functionality of sending and receiving arbitrary string data between chains.
+
+To start, create the function declarations for `sendMessage`, which will send string data cross-chain to the specified destination address' receiver. This function allows callers to specify the destination chain ID, destination address to send to, relayer fees, required gas limit for message execution on destination address, and the actual message data.
```

Comment:
nit: **From** "... for message execution on destination address" **to** "... for message execution at the destination address' receiver" (or just "at the destination").

super nit: Would it be more accurate to call it "message handling" instead of "message execution"? Or maybe "execution of the message data"?

Reply:
Thanks for the suggestions. I think for the purposes of this tutorial, we should just leave it as "message execution". The details of failed message execution and retries are covered in the more in-depth Teleporter documentation.